src/device.ts

```diff
@@ -312,15 +312,24 @@ export class Device extends vscode.Disposable {
      */
     public async mkdir_p(path: string): Promise<void> {
         const names = path.split('/');
+
+        // If path had a preceding slash, remove the empty path segment
```

Comment:
Hmm... we should probably always have a leading slash. I don't think we have a concept of a current working directory when we call this function. We could probably get away without the if and just shift the array. Probably not something to worry about though.

Reply:
Yeah, I figure that it's better to have some safety here rather than causing odd behavior in the future when someone uses this function differently. I don't have any preference as to how we ensure that though.

Reply:
Perhaps it would be a good idea to throw an exception if `!path.posix.isAbsolute(path)`?